src/components/List/ListItem/ListItem.jsx

```diff
@@ -32,7 +32,11 @@ const ListItemPropTypes = {
   expanded: PropTypes.bool,
   value: PropTypes.string.isRequired,
   secondaryValue: PropTypes.string,
-  rowActions: PropTypes.arrayOf(PropTypes.node), // TODO
+  /** either a callback render function or a node */
+  rowActions: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.func,
+  ]),
```

Comment:
Is this to fix the dequal max call size w/ useDeepCompareEffect? I think this could be a good overall solution for #1797, and place a deprecating notice on the usage of array of nodes.

Reply:
No they're separate fixes though I think deprecating notice is a good idea